api/views.py

```diff
@@ -37,3 +39,31 @@ class QuestionCategoryViewSet(viewsets.ReadOnlyModelViewSet):
 class UserStatsViewSet(viewsets.ReadOnlyModelViewSet):
     queryset = MyUser.objects.all()
     serializer_class = UserStatsSerializer
+
+
+class ActionsViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    Query Parameters
+    ?recent - if true, renders the list in recent first order
+    """
+    serializer_class = ActionsSerializer
+    permission_classes = [IsAuthenticated, ]
+
+    def get_queryset(self):
+        user = self.request.user
+        is_recent = self.request.query_params.get('recent', False)
+        if is_recent:
+            return user.actions.all().order_by("-time_modified")
```

Comment:
Ok i just realized that you are not limiting the number of useractions returned
if you are just ordering them i recommend using OrderFilter
https://www.django-rest-framework.org/api-guide/filtering/#orderingfilter
you can also add a limit to the number of user actions returned
instead of filtering it in angular it should be filtered in the back-end for the sake of performance